### PR TITLE
HookArgument cop

### PR DESCRIPTION
**Pull request by backus**
This is extracted from #104. I'm just rebasing, improving the tests a bit using the new matcher, and I'm going to change this to support `EnforcedStyle` and `NodePattern`

### CODE REVIEW

File: lib/rubocop/cop/rspec/hook_argument.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# encoding: utf-8
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/hook_argument.rb]: Can be removed.

File: lib/rubocop/cop/rspec/hook_argument.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Omit the default :each argument for RSpec hooks.
+      #
+      # @example when configuration is `EnforcedStyle: :implicit`
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/hook_argument.rb]: `:implicit` shouldn't be a symbol here, since it's illustrating the config.yml?

Comment (backus) [inline lib/rubocop/cop/rspec/hook_argument.rb]: Correct

File: lib/rubocop/cop/rspec/hook_argument.rb (hunk 1 above)
Comment (backus) [inline lib/rubocop/cop/rspec/hook_argument.rb]: Thanks.

**Pull request merged**